### ISSUE TITLE
Requirement: numpy<2.0.0

### ISSUE THREAD

**Issue by josk0** [IBM/lale#1375]
Should the `numpy<2.0.0` requirement be followed throughout? It _is_ stated in docs/requirements.txt but _not_ in setup.py.

Very new to this, but it seems this creates a potential dependency problem since lale is incompatible with numpy 2.0 (e.g. use of `np.NaN` in lale/lib/xgboost/xgb_classifier.py", line 792)

**Comment by rootsmusic** [comment]
xgboost 2.1.0 is [compatible with numpy 2.](https://github.com/numpy/numpy/issues/26191)0, but I don't believe lale 0.8.2 is.

**Comment by josk0** [comment]
Yes, the wrapper is the/one issue. So requirements maybe should be adjusted. I just don’t know that’s done…

**Comment by hirzel** [comment]
Hi, thank you for raising this issue! Instead of constraining the numpy version everywhere, it might be better to update the XGBoost wrapper to work with numpy 2.0.0. Would you like to try that?

Numpy 2.0.0 was released on PyPI on June 16th. After that, the Lale [tests](https://github.com/IBM/lale/actions/runs/9781406831) from July 3rd succeeded without an upper bound on numpy. And some of the tests explicitly install the [newest](https://github.com/IBM/lale/blob/165a4c0f4b6da0e33df5e62040fe45f34688a662/.github/workflows/build.yml#L292) numpy version, in this case, 2.0.0.

**Comment by rootsmusic** [comment]
> And some of the tests explicitly install the [newest](https://github.com/IBM/lale/blob/165a4c0f4b6da0e33df5e62040fe45f34688a662/.github/workflows/build.yml#L292) numpy version, in this case, 2.0.0.

@hirzel That test pinned numpy for snapml, so I'm assuming that snapml isn't compatible with numpy 2.0 yet?

**Comment by hirzel** [comment]
@rootsmusic The test I linked above ([test_newer](https://github.com/IBM/lale/blob/165a4c0f4b6da0e33df5e62040fe45f34688a662/.github/workflows/build.yml#L292)) does not pin numpy for SnapML. There are some other tests that do, because at the time they were written, SnapML had these restrictions. It is possible that the newest SnapML no longer has these restrictions, which would require an experiment to check.

**Comment by jakirkham** [comment]
Was this already addressed?

Can see the upper bound on NumPy was removed in PR: https://github.com/IBM/lale/pull/1342

**Comment by shinnar** [comment]
yes, the bound was removed and lale is now compatible with numpy 2.
In fact, recent CI builds (e.g. https://github.com/IBM/lale/actions/runs/18785448220/job/53602938283) are testing using numpy 2 (2.2.6 in the linked example)
